docs/configuration/OVERVIEW.md

```diff
@@ -1,14 +1,23 @@
 # Configuration mechanism
 
-The aim of this document is to help developers to implement configuration inside components, in order to be compliant
-with __CMS architecture__. This will give the possibility to __Business Analyst__ to configure the
-application/components.
+The aim of this document is to help developers to implement configurability in a library or an application.
 
-## Component config types
+You may need configurability if you are developing components that can have different behaviors or displays depending on who is using them or how.
+Let's take the example of a library with calendar components used by many projects all around the world.
+Depending on their culture, some users may want a Gregorian Calendar, others may need the Lunar one.
+You will probably want to make your component configurable at least at build time to meet everyone's needs.
+
+Now let's imagine the same use case, not for a library, but for an application. You need to implement some sort of runtime configurability to meet
+everyone's needs. That is the purpose of Otter's Configuration pattern.
+
+In this documentation, you will learn how to implement this pattern and how to extract metadata that will describe how your project can be configured.
+The JSON metadata can be used to build your own override application, or as an input for a CMS plugin, for example.
+
+## Component configuration types
 
 ```
 
-    store config
+  dynamic config
```

Comment:
Not sure the term is correct as they are all "dynamic config".
Maybe we can use "customized configuration"?

But a link scribing the different configuration process (or giving an example of it) can help :S